tests/integration_tests_rust.rs

```diff
@@ -327,3 +327,31 @@ fn do_connection_restart_behavior(persist: bool) {
 		assert!(node_b.list_peers().is_empty());
 	}
 }
+
+#[test]
+fn concurrent_connections_succeed() {
+	let (_bitcoind, electrsd) = setup_bitcoind_and_electrsd();
+	let (node_a, node_b) = setup_two_nodes(&electrsd, false);
+
+	let node_a = Arc::new(node_a);
+	let node_b = Arc::new(node_b);
+
+	let node_id_b = node_b.node_id();
+	let node_addr_b = node_b.listening_addresses().unwrap().first().unwrap().clone();
+
+	std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
Could you add a comment on why the `sleep` is needed? Is there any way to avoid it?

Reply:
Could we have an `is_listening` method, even if only for testing?